File: src/main/scala/com/vc/easy/L766.scala
package com.vc.easy

object L766 {
  def isToeplitzMatrix(matrix: Array[Array[Int]]): Boolean = {
    (1 until matrix.length).foreach(i => {
      (1 until matrix(i).length).foreach(j => {
        if(matrix(i)(j) != matrix(i - 1)(j - 1)) return false
      })
    })
    true
  }
}

File: src/main/java/com/vc/easy/L766.java
package com.vc.easy;

class L766 {
    public boolean isToeplitzMatrix(int[][] matrix) {
        if(matrix == null || matrix.length == 0) return true;

        int n = matrix.length;
        int m = matrix[0].length;

        for(int col = 0; col < m; col++) {
            int row = 0;
            int colInner = col;
            int element = matrix[row][colInner];
            while(row < n && colInner < m) {
                if(element != matrix[row][colInner]) return false;
                row++;
                colInner++;
            }
        }

        for(int row = 1; row < n; row++) {
            int col = 0;
            int rowInner = row;
            int element = matrix[rowInner][col];
            while(rowInner < n && col < m) {
                if(element != matrix[rowInner][col]) return false;
                rowInner++;
                col++;
            }
        }

        return true;
    }
}
